graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/types/generateDirective.kt

```diff
@@ -29,20 +30,27 @@ import kotlin.reflect.KClass
 import kotlin.reflect.KProperty
 import com.expediagroup.graphql.annotations.GraphQLDirective as GraphQLDirectiveAnnotation
 
-internal fun generateDirectives(generator: SchemaGenerator, element: KAnnotatedElement, parentClass: KClass<*>? = null): List<GraphQLDirective> {
+internal fun generateDirectives(
+    generator: SchemaGenerator,
+    element: KAnnotatedElement,
+    location: Introspection.DirectiveLocation,
+    parentClass: KClass<*>? = null
+): List<GraphQLDirective> {
     val annotations = when {
         element is KProperty<*> && parentClass != null -> element.getPropertyAnnotations(parentClass)
         else -> element.annotations
     }
 
     return annotations
         .mapNotNull { it.getDirectiveInfo() }
+        .filter { it.directiveAnnotation.locations.contains(location) }
         .map { getDirective(generator, it) }
 }
 
 internal fun generateFieldDirectives(generator: SchemaGenerator, field: Field): List<GraphQLDirective> =
     field.annotations
         .mapNotNull { it.getDirectiveInfo() }
+        .filter { it.directiveAnnotation.locations.contains(Introspection.DirectiveLocation.ENUM_VALUE) }
```

Comment:
shouldn't this be `DirectiveLocation. FIELD_DEFINITION` or `DirectiveLocation. INPUT_FIELD_DEFINITION`?

Reply:
looks like it is invoked from enum generator -> should we pass it explicitly instead?

Reply:
The name `generateFieldDirectives` is referring to the fact that enum values are of type `java.lang.reflect.Field`. If we did make it an argument it would just always be the value `ENUM_VALUE`

Reply:
this seems misleading then - lets either rename the method name OR make the location type an argument

Reply:
Updated